experimental/traffic-portal/src/app/core/cdns/cdn-table/cdn-table.component.ts

```diff
@@ -0,0 +1,248 @@
+			href: (selectedRow): string => `/core/cdns/${selectedRow.id}/notifications` ,
+			name: "Manage Notifications"
+		},
+	];
+
+	/**
+	 * A subject that child components can subscribe to for access to the fuzzy
+	 * search query text.
+	 */
+	public fuzzySubject: BehaviorSubject<string>;
+
+	/** Form controller for the user search input. */
+	public fuzzControl: FormControl = new FormControl("");
+
+	constructor(
+		private readonly alerts: AlertService,
+		private readonly api: CDNService,
+		public readonly auth: CurrentUserService,
+		private readonly dialog: MatDialog,
+		private readonly route: ActivatedRoute,
+	) {
+		this.fuzzySubject = new BehaviorSubject<string>("");
+		this.cdns = this.api.getCDNs();
+	}
+
+	/** Initializes table data, loading it from Traffic Ops. */
+	public ngOnInit(): void {
+		this.route.queryParamMap.subscribe(
+			m => {
+				const search = m.get("search");
+				if (search) {
+					this.fuzzControl.setValue(decodeURIComponent(search));
+					this.updateURL();
+				}
+			}
+		);
+	}
+
+	/** Update the URL's 'search' query parameter for the user's search input. */
+	public updateURL(): void {
+		this.fuzzySubject.next(this.fuzzControl.value);
+	}
+
+	/**
+	 * Queues or clears updates on a group of CDNs.
+	 *
+	 * @param cdn The CDN on which to queue updates.
+	 * @param queue Whether updates should be queued (`true`) or cleared
+	 * (`false`).
+	 */
+	private async queueUpdates(cdn: ResponseCDN, queue: boolean = true): Promise<void> {
+		const title = `${queue ? "Queue" : "Clear"} Updates on ${cdn.name}?`;
+		const action = queue ? "queue" : "dequeue";
+		const ref = this.dialog.open<DecisionDialogComponent, DecisionDialogData, boolean>(DecisionDialogComponent, {
+			data: {
+				message: `Are you sure you want to ${action} server updates for all the ${cdn.name} servers?`,
+				title,
+			}
+		});
+		if (!await ref.afterClosed().toPromise()) {
+			return;
+		}
+		await this.api.queueCDNUpdates(cdn, action);
+		this.alerts.newAlert(
+			AlertLevel.SUCCESS,
+			"Queued CDN server updates",
+		);
+	}
+
+	/**
+	 * Asks the user for confirmation before deleting a CDN.
+	 *
+	 * @param cdn The CDN (potentially) being deleted.
+	 */
+	private async delete(cdn: ResponseCDN): Promise<void> {
+		const ref = this.dialog.open<DecisionDialogComponent, DecisionDialogData, boolean>(DecisionDialogComponent, {
+			data: {
+				message: `Are you sure you want to delete the ${cdn.name} CDN?`,
+				title: `Delete ${cdn.name}`
+			}
+		});
+		if (await ref.afterClosed().toPromise()) {
+			await this.api.deleteCDN(cdn);
+			this.cdns = this.api.getCDNs();
+		}
+	}
+
+	/**
+	 * Handles a context menu event.
+	 *
+	 * @param a The action selected from the context menu.
+	 */
+	public handleContextMenu(a: ContextMenuActionEvent<ResponseCDN>): void {
+		switch(a.action) {
+			case "queue":
+				this.queueUpdates(a.data as ResponseCDN);
+				break;
+			case "dequeue":
+				this.queueUpdates(a.data as ResponseCDN, false);
```

Comment:
in general `as` should be avoided whenever possible. In these cases, I'd just check for an array of data like you do for the `delete` action. Or, if it's intended to handle multiple CDN queues at a time, it should handle that.

Reply:
Avoiding `as` in 2f6c6c9ea5